api/rpm_and_image_fetcher.py

```diff
@@ -0,0 +1,72 @@
+import os
+import requests
+import logging
+import json
+
+# Set your GitHub access token as an environment variable
+GITHUB_TOKEN = os.environ.get("GITHUB_PERSONAL_ACCESS_TOKEN")
+headers = {
+    "Authorization": f"token {GITHUB_TOKEN}"
+}
+
+REPO_OWNER = "openshift-eng"
+REPO_NAME = "ocp-build-data"
+
+def get_directory_contents(branch, directory):
+    path = f"{REPO_OWNER}/{REPO_NAME}/contents/{directory}?ref={branch}"
+    contents_api = f"https://api.github.com/repos/{path}"
+    response = requests.get(contents_api, headers=headers)
+    response.raise_for_status()  # Raise an exception if the request fails
+    return response
+
+def fetch_data():
+    result = []
+
+    # Separate entry for the "openshift-3.11" branch
+    branch_name = "openshift-3.11"
+    logging.debug(f"Processing branch {branch_name}...")
+
+    try:
+        rpms_content = get_directory_contents(branch_name, "rpms")
+        images_content = get_directory_contents(branch_name, "images")
+
+        if rpms_content.status_code == 200 and images_content.status_code == 200:
+            rpms = [rpm["name"] for rpm in rpms_content.json() if rpm["type"] == "file"]
+            images = [image["name"] for image in images_content.json() if image["type"] == "file"]
+
+            result.append({
+                "branch": branch_name,
+                "rpms": rpms,
+                "images": images,
+            })
+
+    except requests.exceptions.HTTPError as e:
```

Comment:
The try-except block contains not just the API call right, there can be other exceptions like `KeyError` that can arise. If we're handling it, it might be good idea to think about what all exceptions that can pop up, not just the `HTTPError`.

Reply:
Actually it might not be a good idea to catch the error to just to log it. If the server crashes, we'll exactly know which code is causing the issue. So unless we need to do something meaningful when an exception occurs, we can just let it crash.

Reply:
It took me few days but I understand what you mean now :D removing try block now.